packages/core/src/Synchronizer.ts

```diff
@@ -589,6 +595,54 @@ export class Synchronizer extends EventEmitter {
         return preimages
     }
 
+    async genSealedEpochProof(
+        options: {
+            epoch?: bigint
+            attesterId?: bigint
+            preimages?: bigint[]
+        } = {}
+    ): Promise<BuildOrderedTree> {
+        const attesterId =
+            options.attesterId?.toString() ?? this.attesterId.toString()
+        const unsealedEpoch = await this._db.findOne('Epoch', {
+            where: {
+                sealed: false,
+                number: options.epoch ? Number(options.epoch) : undefined,
+                attesterId,
+            },
+            orderBy: {
+                epoch: 'asc',
+            },
+        })
+
+        if (!unsealedEpoch) {
+            throw new Error(`Synchronizer: sealing epoch is not required.`)
+        }
+        const attestation = await this._db.findOne('Attestation', {
+            where: {
+                attesterId: attesterId.toString(),
+                epoch: unsealedEpoch.number,
+            },
+        })
+        if (!attestation) {
+            throw new Error(
+                `Synchronizer: no attestation is made in epoch ${unsealedEpoch.number}.`
+            )
+        }
+        const epoch = options.epoch ?? unsealedEpoch.number
+        const preimages =
+            options.preimages ??
+            (await this.genEpochTreePreimages(epoch, attesterId))
+        const { circuitInputs } =
+            BuildOrderedTree.buildInputsForLeaves(preimages)
```

Comment:
Make sure to pass `settings.epochTreeArity`, `settings.epochTreeDepth` and `settings.fieldCount` so it doesn't use the default config.